UndertaleModLib/Decompiler/Instructions/Decompiler.ExpressionTwo.cs

```diff
@@ -34,15 +34,69 @@ public override Statement CleanStatement(DecompileContext context, BlockHLStatem
             return this;
         }
 
-        public override string ToString(DecompileContext context)
+        private string AddParensIfNeeded(Expression argument, DecompileContext context)
+        {
+            string arg = argument.ToString(context);
+            bool needsParens;
+            if (arg[0] != '(' &&
+                argument is not ExpressionConstant &&
+                arg.Contains(' ', StringComparison.InvariantCulture))
+                needsParens = true;
+            else
+                needsParens = false;
+
+
+            if (argument is ExpressionTwo argumentAsBinaryExpression)
+            {
+                int outerPriorityLevel = Opcode switch
+                {
+                    UndertaleInstruction.Opcode.Mul or UndertaleInstruction.Opcode.Div => 2,
+                    UndertaleInstruction.Opcode.Add or UndertaleInstruction.Opcode.Sub => 1,
+                    _ => 0,
+                };
+
+                // First, no parentheses on this type
+                arg = argumentAsBinaryExpression.ToStringNoParens(context);
+
+                int argPriorityLevel = argumentAsBinaryExpression.Opcode switch
+                {
+                    UndertaleInstruction.Opcode.Mul or UndertaleInstruction.Opcode.Div => 2,
+                    UndertaleInstruction.Opcode.Add or UndertaleInstruction.Opcode.Sub => 1,
+                    _ => 0,
+                };
+
+
+                // Suppose we have "(arg1a argOp arg1b) opcode argument2", and are wondering whether the depicted parentheses are needed
+                // If the argument's opcode is more highly-prioritized than our own, such as it being multiplication
+                // while we use addition, then no parentheses are required.
+                // If the argument's opcode doesn't fall into typical math rules (that is, I don't know my full order of operations)
+                // Assume it has lower priority and needs parentheses to clarify.
+                // Parentheses are also not needed for operations of the same level, especially string concatenation.
+                needsParens = (outerPriorityLevel > argPriorityLevel);
+                if (outerPriorityLevel == 0)
+                    needsParens = true; // Better safe than sorry
```

Comment:
This is how I'd fix the above comment (non-commutative operators):
```suggestion
                    needsParens = true; // Better safe than sorry
                
                // Non-commutative operators may still need parentheses, e.g `a - (b - c)`
                bool nonCommutative = Opcode == UndertaleInstruction.Opcode.Sub || Opcode == UndertaleInstruction.Opcode.Div;
                if (isSecond && nonCommutative && Opcode == argumentAsBinaryExpression.Opcode)
                    needsParens = true;
```

Reply:
Wait whoops I forgot that I added an additional argument to the function in my fork which I forgot to put here oops sorry.